src/Avatar.ts

```diff
@@ -112,6 +114,46 @@ export function defaultAvatarUrlForString(s: string): string {
     return dataUrl;
 }
 
+/**
+ * If client is particular about role colors  user role/user directory category then
+ * approach below would be sufficient to replace colors configured in custom config as well as color
+ * You can configure your own color in 'avatarColors' keypair in config file by type of role category
+ * Not having this config or not having color in config parameter would default
+ * avatar colors back to matrix color
+ * configAvatarColors needs to be passed from config and read via config.json + config.ts files
+ * configAvatarColors needs to be obj{key:value} pair type in config
+ * @param {string} s
+*/
+function getSpecificColorFromConfig(s: string): string {
+    let configuredAvatarColorsKeyPair = config.avatarColors;
+    let selectedAvatarColor = '';
+    /**
+     * Key (s) needs to find a match in config keypair value stored in "avatarColors"
+     * If matching value is found then just use that value otherwise go to next step where
+     * condition will apply for non matching role category
+     */
+    let selectedAvatarColorFromConfig = configuredAvatarColorsKeyPair[s];
+    if (selectedAvatarColorFromConfig) {
+        selectedAvatarColor = selectedAvatarColorFromConfig;
+    }
+    /**
+     * If no color key matches form given 's' to keypair in config.json
+     * then make color stable for avatar by calculating index based on
+     * length of 's' string passed in. For all non matching name key,
```

Comment:
Please updated the comment to match the implementation e.g. replace
```
     * length of 's' string passed in. For all non matching name key,
     * select color based on name length which will never fail.
```
with
```
     * content of 's' string passed in.
```

Reply:
Too easy suggestion has been implemented.